packages/react-devtools-core/src/backend.js

```diff
@@ -8,60 +8,134 @@
  *
  * @flow
  */
-
 'use strict';
 
-// This object is set up by websocketConnect() that is called from ./embed.js.
-declare var FOR_BACKEND: {
-  wall: Object;
-  resolveRNStyle: () => void;
+type ConnectOptions = {
+  host?: string,
+  port?: number,
+  resolveRNStyle?: (style: number) => ?Object,
 };
 
+// TODO: why?
 if (!window.performance) {
   window.performance = {
     now: () => Date.now(),
   };
 }
 
-var installGlobalHook = require('../../../backend/installGlobalHook.js');
-installGlobalHook(window);
-
 var Agent = require('../../../agent/Agent');
 var Bridge = require('../../../agent/Bridge');
+var installGlobalHook = require('../../../backend/installGlobalHook.js');
 var inject = require('../../../agent/inject');
 var setupRNStyle = require('../../../plugins/ReactNativeStyle/setupBackend');
 var setupRelay = require('../../../plugins/Relay/backend');
 
-FOR_BACKEND.wall.onClose(() => {
-  if (agent) {
-    agent.emit('shutdown');
+installGlobalHook(window);
+
+if (window.document) {
+  window.__REACT_DEVTOOLS_GLOBAL_HOOK__.on('react-devtools', agent => {
+    var setupHighlighter = require('../../../frontend/Highlighter/setup');
+    setupHighlighter(agent);
+  });
+}
+
+function connectToDevTools(options: ?ConnectOptions) {
+  var {
+    host = 'localhost',
+    port = 8097,
+    resolveRNStyle = null,
+  } = options || {};
+
+  var messageListeners = [];
+  var closeListeners = [];
+  var uri = 'ws://' + host + ':' + port;
+  var ws = new window.WebSocket(uri);
+  ws.onclose = handleClose;
+  ws.onerror = handleClose;
+  ws.onmessage = handleMessage;
+  ws.onopen = function() {
+    var wall = {
+      listen(fn) {
+        messageListeners.push(fn);
+      },
+      onClose(fn) {
+        closeListeners.push(fn);
+      },
+      send(data) {
+        ws.send(JSON.stringify(data));
+      },
+    };
+    setupBackend(wall, resolveRNStyle);
+  };
+
+  var hasClosed = false;
+  function handleClose() {
+    if (!hasClosed) {
+      hasClosed = true;
+      setTimeout(() => {
+        connectToDevTools(options);
+      }, 2000);
```

Comment:
This was intentional change from RN team because they had some issues with 200ms reconnection (it was making things slow or something). 500ms is better than 200ms but I don't want to touch it without understanding the issues they had in the first place. Maybe @yungsters or @javache can provide more details.